File: cpp/204_计数质数_easy.cpp
class Solution {
public:
    int countPrimes(int n) {
        int ans = 0;
        vector<int> temp(n, 1);
        for(int i=2; i<n; i++)
            if(temp[i]){
                ans ++;
                if((long long )i*i<n)
                    for(int j=i*i; j<n; j+=i)
                        temp[j] = 0;
            }
        return ans;
    }
};

File: python/204_计数质数_easy.py
class Solution:
    def countPrimes(self, n: int) -> int:
        ans = 0
        a = [1]*n
        for i in range(2, n):
            if a[i]:
                ans += 1
                if i*i<n:
                    for i in range(i*i, n, i):
                        a[i] = 0
        return ans